src/Tuple.mo

```diff
@@ -0,0 +1,116 @@
+/// Contains modules for working with tuples of different sizes.
+
+import Types "Types";
+
+module {
+
+  /// Swaps the elements of a tuple.
+  ///
+  /// ```motoko
+  /// import Tuples "mo:new-base/Tuple";
+  /// assert Tuple.Tuple2.swap((1, "hello")) == ("hello", 1);
+  /// ```
+  public func swap<A, B>((a, b) : (A, B)) : (B, A) = (b, a);
+
+  /// Creates a textual representation of a tuple for debugging purposes.
+  ///
+  /// ```motoko
+  /// import Tuple "mo:new-base/Tuple";
+  /// import Nat "mo:new-base/Nat";
+  /// assert Tuple.toText((1, "hello"), Nat.toText, func (x: Text): Text = x) == "(1, hello)";
+  /// ```
+  public func toText<A, B>(t : (A, B), toTextA : A -> Text, toTextB : B -> Text) : Text = "(" # toTextA(t.0) # ", " # toTextB(t.1) # ")";
```

Comment:
I like the idea of including a `Tuple` module, although maybe we can rearrange this to use a submodule for each number of elements? For instance:

```motoko
module Tuple {
  module Tuple2 {
    public func toText<A, B>((a, b) : (A, B)) { ... }
  };
  module Tuple3 {
    public func toText<A, B, C>((a, b, c) : (A, B, C)) { ... }
  };
}
```

This way, we could keep consistent function naming, and it would also be possible to import these functions in a reasonably convenient way:

```motoko
import { Tuple2, Tuple3 } "mo:base/Tuple";
```

We could maybe also use code generation to generate all of these from templates. Since there are a lot of directions we could go with this, maybe we could move this into a separate PR for now?